RFC/src/RFC-0170_NetworkCommunicationProtocol.md

```diff
@@ -0,0 +1,316 @@
+
+A CN that is also a VN should maintain more peer connections with other VNs, but also have some connections with BNs.
+CNs that are part of Validator Node committees should attempt to maintain permanent connections with the other members of the committee to ensure that quick consensus can be achieved.
+
+To maintain connections with peers, the following process can be performed.
+Discover peers using discovery requests, and add their details to the local routing table.
+The CN can decide how the peer connections should be selected from the routing table by either:
+ - manually selecting a subset,
+ - automatically selecting a random subset or
+ - selecting a subset of neighbouring nodes with similar node IDs. 
+
+Maintaining communication channels is important and the following process can be followed in an attempt to keep peer connections alive:
+For an existing peer connection.
+When more than 30 minutes have passed since the last communication with that peer, then a heartbeat message should be sent to the peer in an attempt to keep the connection with the peer alive.
+If the peer connection was not established for a specific purpose, such as with the connections between committee members, then a new replacement peer can be selected from the local routing table.
+A new connection can then be established and maintained between the current node and the newly selected node.
+If that specific connection is important, such as with the connections between committee members, then the current CN or CC must wait and attempt to create a new connection with that same peer.
+If more than 90 minutes have passed since the last successful communication with the peer node, a new discovery request can be sent on the Tari communication network in an attempt to locate that peer again.
+Losing of a peer might happen in cases where the CN or CC went temporarily offline and their dynamic communication address changed, requiring the discovery process to be performed again before a direct P2P communication channel can be established.
+
+#### Functionality Required of Consensus Nodes
+
+- It MUST select a cryptographic key pair used for identification on the Tari Communication network.
+- A CN MAY request the peer addresses of CNs with similar node IDs from other CNs to extend their local routing table. 
+- If a CN is a VN, then a node ID MUST be obtained by registering on the Base layer.
+- If a CN is a BN, then a node ID MUST be derived from the nodes identification public key.
+- A new CN MUST submit a joining request to the Tari communication network so that the nodes peer address can be added to the routing table of neighbouring peers in the network.
+- If a CN receives a new joining request with a similar node ID (within a network selected threshold), then the peer address specified in the joining request MUST be added to its local routing table.
+- When a CN receives an encrypted message, the node MUST attempt to open the message.
```

Comment:
This bears some thinking about, recently heard about attacks where a malicious message is constructed and the process of trying to open it can crash a node. Might just be hypothetical but I do think there is a spam vector here.

Reply:
Maybe the source public key must always be in the clear so that a check can be performed to determine if it was blacklisted before attempting to open the message.